nomad/resource_job_fields.go

```diff
@@ -0,0 +1,1081 @@
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"file": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+						},
+					},
+				},
+				"lifecycle": {
+					Type:     schema.TypeList,
+					Optional: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"hook": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"sidecar": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+						},
+					},
+				},
+				"logs":      getLogsFields(),
+				"resources": getResourcesFields(),
+				"service":   getServiceFields(),
+				"template": {
+					Type:     schema.TypeList,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"change_mode": {
+								Type:     schema.TypeString,
+								Default:  "restart",
+								Optional: true,
+							},
+							"change_signal": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"data": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"destination": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"env": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+							"left_delimiter": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"perms": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"right_delimiter": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"source": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"splay": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"vault_grace": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+						},
+					},
+				},
+				"vault": getVaultFields(),
+				"volume_mount": {
+					Type:     schema.TypeList,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"volume": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"destination": {
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+							"read_only": {
+								Type:     schema.TypeBool,
+								Optional: true,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getNetworkFields() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"mbits": {
+					Type:     schema.TypeInt,
+					Optional: true,
+				},
+				"mode": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"port": {
+					Type:     schema.TypeList,
+					Optional: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
```

Comment:
this will need a `label` field as well... that is the name of the port resource.